core/fetcher.go

```diff
@@ -26,6 +26,21 @@ func NewBlockFetcher(client Client) *BlockFetcher {
 	}
 }
 
+// GetBlockInfo queries Core for additional block information, like Commit and ValidatorSet.
+func (f *BlockFetcher) GetBlockInfo(ctx context.Context, height *int64) (*types.Commit, *types.ValidatorSet, error) {
+	commit, err := f.Commit(ctx, height)
+	if err != nil {
+		return nil, nil, fmt.Errorf("core/fetcher: getting commit: %w", err)
+	}
+
+	valSet, err := f.ValidatorSet(ctx, height)
```

Comment:
Is this guaranteed to be the same height as the call to `f.Commit`, if `height` is `nil`?

Reply:
No guarantees. Good catch!

Reply:
Should be commit.Height instead.

Reply:
Does `f.ValidatorSet` error if height is nil?

Reply:
IIRC if `height` is `nil` then it means "latest." If that's the case, then I don't see any reason why `f.ValidatorSet()` would error when requesting the latest validator set. Might be missing something though?

Reply:
Not it does not error if nil. Nil is for the latest as John said

Reply:
> Is this guaranteed to be the same height as the call to f.Commit, if height is nil?

So the difference in heights for commit and valset would only occur in the case that between requesting the latest commit and the latest valset a new block "happened". Yeah, better to use commit.Height as Hlib said then.

Reply:
very good catch thankie @adlerjohn :)

Reply:
Just noting that we always call this method with a concrete `&block.Height` taken from a block that was received via a new block event. So this will never be called with latest/`nil` as far as I understand but always with a concrete height here. If that is actually the expectation of how GetBlockInfo is to be used, it would be even better by making this explicit and to error on `nil` instead.

Reply:
If don't need to pass nil there, then we don't need to have a ptr to int
> it would be even better by making this explicit and to an error on nil instead.

Reply:
But for consistency, we can keep a ptr still, but make it work with nil as suggested. If don't need this now, it does not mean we won't need it at all.

Reply:
https://github.com/celestiaorg/celestia-node/issues/336

resolving

Reply:
> If don't need this now, it does not mean we won't need it at all.

Yeah, but it's better to introduce things as they are needed IMO instead of deciding APIs for some hypothetical future that might or might not ever occur.